packages/panzer/CMakeLists.txt

```diff
@@ -28,10 +28,9 @@ SET(${PACKAGE_NAME}_ADD_EXPENSIVE_CUDA_TESTS
   "Add expensive convergence tests to cuda builds.  Not recommended for CMAKE_BUILD_TYPE=DEBUG and CUDA")
 MESSAGE(STATUS "Convergence tests: ${${PACKAGE_NAME}_ADD_EXPENSIVE_CUDA_TESTS}")
 
-TRIBITS_ADD_OPTION_AND_DEFINE(${PACKAGE_NAME}_ENABLE_FADTYPE
-  PANZER_FADTYPE
-  "Choose the Sacado automatic differentiation scalar type (default is DFad)."
-  "Sacado::Fad::DFad<RealType>" )
+SET(${PACKAGE_NAME}_ENABLE_FADTYPE  "Sacado::Fad::DFad<RealType>"
```

Comment:
Definition of enable
transitive verb

1 a : to provide with the means or opportunity
      // training that enables people to earn a living
  b : to make possible, practical, or easy
      // a deal that would enable passage of a new law
  c : to cause to operate
      // software that enables the keyboard
2 : to give legal power, capacity, or sanction to
    // a law enabling admission of a state

So in this case none of the definitions apply after our changes. Other uses of ENABLE in Trilinos/Tribits are booleans (ON/OFF) matching definition 1c. After these changes we are really providing or overriding a default (there is no option to disable this - never really was.) I know this means changing the various cmake specs stored in the repo and possibly disrupting developers who have stored configs, but I think the clarity is worth it.  

A similar item would be Tpetra_DefaultNode which is not a good example (I hear complaints about that one regularly.)

Perhaps ${PACKAGE_NAME}_DEFAULT_FADTYPE ? There is not a good situation here with existing macros and conventions. Perhaps I should just live with the cognitive dissonance in order to minimize disruption.

This won't merge until Ross removes his change request so give me feedback on this naming question.

Thanks!

Reply:
Removing the term ENABLE is fine. ${PACKAGE_NAME}_FADTYPE is probably the best long term solution.  I think Ross is worried about backwards compatibility so if we make this change, we need to modify all the atdm scripts stored in trilinos. That should be a simple search and relpace. I would prefer not to use "${PACKAGE_NAME}_DEFAULT_FADTYPE as this implies that it is only the default and that users can change it. They can't. It is s single configure time option they are stuck with. The Tpetra_DefaultNode makes sense since multiple node types can be enabled at compile time. If a user doesn't explicitly specify a node type in the template arguments, it defaults to that one. There are apps that only use the default (never explicitly specify the node type in their code) and there are apps that use multiple node types at the same time and are explicit in specifying node types in all use of tpetra.